File: JAVA/Split an array into two equal Sum subarrays.java
class Solution {
    public boolean canSplit(int arr[]) {
        int arrSum = 0, sum = 0, n = arr.length;
        for(int i=0; i < n; i++) {
            arrSum += arr[i];
        }
        for(int i=0; i < n; i++) {
            sum += arr[i];
            if(sum * 2 == arrSum) {
                return true;
            }
        }
        return false;
    }
}

File: C++/Split an array into two equal Sum subarrays.cpp
class Solution {
  public:
    bool canSplit(vector<int>& arr) {
      int n=arr.size();
      int i=0;
      int j=n-1;
      int leftsum=arr[0];
      int rightsum=arr[n-1];
      while(i<j){
          if(leftsum==rightsum && i+1==j){
              return true;
          }
          else if(leftsum>rightsum){
              j--;
              rightsum+=arr[j];
          }
          else{
                 i++;
             leftsum+=arr[i];
          }
      }
      return false;
    }
};
